library/aur.py

```diff
@@ -253,13 +256,15 @@ def main():
     else:
         use = params['use']
 
-    if params['upgrade'] and (params['name'] or use == 'makepkg'):
-        module.fail_json(msg="Upgrade cannot be used with this option.")
+    if params.get('upgrade', False) and use == 'makepkg':
+        module.fail_json(msg="Upgrade cannot be used with the helper 'makepkg'.")
+
+    if module.check_mode:
+        check_packages(module, params['name'])
+    elif params.get('upgrade', False):
```

Comment:
why are you using get for upgrade and not for the other booleans?

Reply:
With the change discussed above to remove the default for `upgrade`, this property might not be in the `params` dict. So to make sure we handle that default correctly, we could either insert a `False` into the dict if we see that it's not there, or we could use `dict.get` to resolve the default after a failed lookup.

Reply:
all right